substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/thread/VMThreads.java

```diff
@@ -680,6 +682,45 @@ public static int getNewThreadStatus(CFunctionOptions.Transition transition) {
         }
     }
 
+    /**
+     * A thread-local enum conveying any actions needed before thread begins executing Java code.
+     */
+    public static class TransitionToJavaSupport {
+
+        /** The actions to be performed. */
+        private static final FastThreadLocalInt actionTL = FastThreadLocalFactory.createInt();
+
+        /** The thread does not need to take any action. */
+        private static final int NO_ACTION = 0;
+        /** Code synchronization should be performed due to newly installed code. */
+        private static final int SYNCHRONIZE_CODE = NO_ACTION + 1;
+
+        @Uninterruptible(reason = "Called from uninterruptible code.", mayBeInlined = true)
+        public static boolean isActionPending() {
+            return actionTL.getVolatile() != NO_ACTION;
+        }
+
+        @Uninterruptible(reason = "Called from uninterruptible code.", mayBeInlined = true)
+        public static void clearActions() {
+            actionTL.setVolatile(NO_ACTION);
+        }
+
+        @Uninterruptible(reason = "Called from uninterruptible code.", mayBeInlined = true)
+        public static void setSynchronizeCode(IsolateThread vmThread) {
+            actionTL.setVolatile(vmThread, SYNCHRONIZE_CODE);
+        }
+
+        public static void requestAllThreadsSynchronizeCode() {
```

Comment:
Yes. This will be called by my new patch in #1977. However, I haven't updated that patch yet because I thought #1903, which it relies on, was about to be merged, and was waiting all that to happen